manifests/config.pp

```diff
@@ -59,6 +59,38 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd  = "/usr/sbin/pam-config -q --mkhomedir | grep session:"
+      $pamconfig_check_cmd  = "/usr/sbin/pam-config -q --sss | grep session:"
+      $nsconfig_check_cmd = "/usr/bin/test $( /bin/grep -cE '(passwd|group|sudoers).*sss' /etc/nsswitch.conf ) -gt 2"
+      $nsconfig_cmd = "/bin/sed -i -e '/^\(passwd\|group\|sudoers\):/{/sss\b/ b;s/^\(.*\)$/\1 sss/}' /etc/nsswitch.conf"
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless      => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless      => $pamconfig_check_cmd,
+      }
+
+      exec { 'echo "sudoers: file sss" >> /etc/nsswitch.conf':
```

Comment:
I'm going to suggest that you not manage nsswitch in the sssd module and instead use the excellent supported module for this...
https://forge.puppet.com/trlinkin/nsswitch

A module should do one thing (only one thing) well.
I suspect many are using the nsswitch module and tampering with nsswitch in here may interfere....